src/mjolnir/timeparsing.cc

```diff
@@ -163,84 +237,44 @@ std::vector<uint64_t> get_time_range(const std::string& str) {
-
-        if (RegexFound(condition, regex)) {
-          condition = FormatCondition(condition, regex, "$1#$2#$3");
-          // fifth is the equivalent of last week in month (-1)
-          condition = FormatCondition(condition, std::regex("\\[-1\\]"), "[5]");
+        // Dec Su[-1] => Dec#Su#[5]
+        if (RegexFound(condition, kWeekdayOfTheMonth.first)) {
+          condition = FormatCondition(condition, kWeekdayOfTheMonth);
+          condition = FormatCondition(condition, kLastWeekday);
         } else {
 
-          regex = std::regex("(?:(Monday|Tuesday|Wednesday|Thursday|Friday|Saturday|"
-                             "Sunday|Mon|Mo|Tues|Tue|Tu|Weds|Wed|We|Thurs|Thur|Th|Fri|Fr|Sat|Sa|Sun|"
-                             "Su)(\\[-?[0-9]\\]))",
-                             std::regex_constants::icase);
-
-          if (RegexFound(condition, regex)) {
-            condition = FormatCondition(condition, regex, "$1#$2");
-            // fifth is the equivalent of last week in month (-1)
-            condition = FormatCondition(condition, std::regex("\\[-1\\]"), "[5]");
+          // Mon[-1] => Mon#[5], Tue[2] => Tue#[2]
+          if (RegexFound(condition, kWeekdayOfEveryMonth.first)) {
+            condition = FormatCondition(condition, kWeekdayOfEveryMonth);
+            condition = FormatCondition(condition, kLastWeekday);
           } else {
 
-            // Feb 16-Oct 15 09:00-18:30
-            regex = std::
-                regex("(?:(January|February|March|April|May|June|July|"
-                      "August|September|October|November|December|Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|"
-                      "Sep|Sept|Oct|Nov|Dec)) (\\d{1,2})",
-                      std::regex_constants::icase);
-
-            if (RegexFound(condition, regex)) {
-              condition = FormatCondition(condition, regex, "$1#$2");
+            // Feb 16-Oct 15 09:00-18:30 => Feb#16-Oct#15 09:00-18:30
+            if (RegexFound(condition, kMonthDay.first)) {
+              condition = FormatCondition(condition, kMonthDay);
             } else {
-              // Feb 2-14
-              regex = std::
-                  regex("(?:(January|February|March|April|May|June|July|"
-                        "August|September|October|November|December|Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|"
-                        "Sep|Sept|Oct|Nov|Dec)) (\\d{1,2})-(\\d{1,2})",
-                        std::regex_constants::icase);
-
-              if (RegexFound(condition, regex)) {
-                condition = FormatCondition(condition, regex, "$1#$2-$1#$3");
+
+              // Feb 2-14 => Feb#2-Feb#14
+              if (RegexFound(condition, kRangeWithinMonth.first)) {
+                condition = FormatCondition(condition, kRangeWithinMonth);
+              } else {
+
+                // Nov - Mar => Nov-Mar
+                if (RegexFound(condition, kMonthRange.first)) {
+                  condition = FormatCondition(condition, kMonthRange);
+                }
```

Comment:
This condition is heavily used in Lithuania to annotate 110 kph speed limit during winter (130 otherwise) -https://en.wikipedia.org/wiki/Speed_limits_in_Lithuania
For specific examples see any A1 way in this relation - https://www.openstreetmap.org/relation/2895620